addons/mail/static/src/js/onchange_on_keydown.js

```diff
@@ -0,0 +1,65 @@
+/** @odoo-module **/
+
+import { patch } from "@web/core/utils/patch";
+import { debounce } from "@web/core/utils/timing";
+import { CharField } from "@web/views/fields/char/char_field";
+import { TextField } from '@web/views/fields/text/text_field';
+import { archParseBoolean } from "@web/views/utils";
+
+const { useEffect } = owl;
+
+/**
+ * Support a key-based onchange in text fields.
+ * The triggerOnChange method is debounced to run after given debounce delay
+ * (or 2 seconds by default) when typing ends.
+ *
+ */
+const onchangeOnKeydownMixin = {
+    setup() {
+        this._super(...arguments);
+
+        if (this.props.onchangeOnKeydown) {
+            const input = this.input || this.textareaRef;
+            this.triggerOnChange = debounce(this.triggerOnChange, this.props.keydownDebounceDelay);
```

Comment:
`triggerOnChange` is used only when the option is true ?

If yes, I would not modify `this.triggerOnChange` and make a variable instead

```js
const triggerOnChange = debounce(this.triggerOnChange, this.props.keydownDebounceDelay);
```